### ISSUE TITLE
[Paper Note] Fast Model Editing at Scale, Eric Mitchell+, ICLR'22

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#2055]
# URL
- https://arxiv.org/abs/2110.11309
# Authors
  - Eric Mitchell
  - Charles Lin
  - Antoine Bosselut
  - Chelsea Finn
  - Christopher D. Manning
# Abstract
  - While large pre-trained models have enabled impressive results on a variety of downstream tasks, the largest existing models still make errors, and even accurate predictions may become outdated over time. Because detecting all such failures at training time is impossible, enabling both developers and end users of such models to correct inaccurate outputs while leaving the model otherwise intact is desirable. However, the distributed, black-box nature of the representations learned by large neural networks makes producing such targeted edits difficult. If presented with only a single problematic input and new desired output, fine-tuning approaches tend to overfit; other editing algorithms are either computationally infeasible or simply ineffective when applied to very large models. To enable easy post-hoc editing at scale, we propose Model Editor Networks using Gradient Decomposition (MEND), a collection of small auxiliary editing networks that use a single desired input-output pair to make fast, local edits to a pre-trained model's behavior. MEND learns to transform the gradient obtained by standard fine-tuning, using a low-rank decomposition of the gradient to make the parameterization of this transformation tractable. MEND can be trained on a single GPU in less than a day even for 10 billion+ parameter models; once trained MEND enables rapid application of new edits to the pre-trained model. Our experiments with T5, GPT, BERT, and BART models show that MEND is the only approach to model editing that effectively edits the behavior of models with more than 10 billion parameters. Code and data available at https://sites.google.com/view/mend-editing.
# Translation (by gpt-4o-mini)
- 大規模な事前学習モデルは、さまざまな下流タスクで印象的な結果をもたらしていますが、既存の最大のモデルでも依然としてエラーを犯すことがあります。また、正確な予測であっても、時間が経つにつれて古くなる可能性があります。トレーニング時にすべての失敗を検出することは不可能であるため、開発者やエンドユーザーが不正確な出力を修正できるようにしつつ、モデル自体はそのままにしておくことが望まれます。しかし、大規模なニューラルネットワークが学習する表現の分散したブラックボックス的な性質は、ターゲットを絞った編集を行うことを難しくしています。単一の問題のある入力と新しい望ましい出力のみが与えられた場合、ファインチューニングアプローチは過剰適合しがちです。他の編集アルゴリズムは、非常に大規模なモデルに適用すると計算上実行不可能であったり、単に効果がないことが多いです。スケールでの容易な事後編集を可能にするために、我々は勾配分解を用いたモデル編集ネットワーク（MEND）を提案します。これは、単一の望ましい入力-出力ペアを使用して、事前学習モデルの動作に対して迅速かつ局所的な編集を行う小さな補助編集ネットワークのコレクションです。MENDは、標準的なファインチューニングによって得られた勾配を変換することを学習し、勾配の低ランク分解を使用してこの変換のパラメータ化を扱いやすくします。MENDは、10億以上のパラメータを持つモデルでも、1台のGPUで1日未満でトレーニング可能です。トレーニングが完了すると、MENDは事前学習モデルへの新しい編集の迅速な適用を可能にします。T5、GPT、BERT、BARTモデルを用いた実験により、MENDが10億以上のパラメータを持つモデルの動作を効果的に編集できる唯一のアプローチであることが示されました。コードとデータは、https://sites.google.com/view/mend-editing で入手可能です。
# Summary (by gpt-4o-mini)
- MEND（モデル編集ネットワーク）は、事前学習モデルの動作を迅速かつ局所的に編集するための手法で、単一の入力-出力ペアを用いて勾配分解を活用します。これにより、10億以上のパラメータを持つモデルでも、1台のGPUで短時間でトレーニング可能です。実験により、MENDが大規模モデルの編集において効果的であることが示されました。

**Comment by AkihikoWatanabe** [comment]
OpenReview:https://openreview.net/forum?id=0DcZxeWfOPt
